clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -1093,16 +1068,17 @@ invariant_checker_t::print_results()
 std::string
 invariant_checker_t::check_for_pc_discontinuity(
     per_shard_t *shard, const per_shard_t::instr_info_t &prev_instr_info,
-    const per_shard_t::instr_info_t &cur_instr_info, bool expect_encoding,
+    const per_shard_t::instr_info_t &cur_memref_info, bool expect_encoding,
     bool at_kernel_event)
 {
     const memref_t prev_instr = prev_instr_info.memref;
     std::string error_msg = "";
     bool have_branch_target = false;
     addr_t branch_target = 0;
-    addr_t prev_instr_trace_pc = prev_instr.instr.addr;
-    const addr_t cur_pc = at_kernel_event ? cur_instr_info.memref.marker.marker_value
-                                          : cur_instr_info.memref.instr.addr;
+    const addr_t prev_instr_trace_pc = prev_instr.instr.addr;
+    // cur_memref_info is a marker (not an instruction) if at_kernel_event is true.
```

Comment:
(I preferred passing in the cur_pc with the logic in the caller where it seems natural and not needing this logic inside here but not strongly enough to request a change; the at_kernel_event is still needed in any case.)